src/pivotal-ui/components/forms/forms.scss

```diff
@@ -1475,14 +1475,49 @@ For an organization name input:
 ---
 title: Constraint Validation
 name: form_constraint_validation
-parent: 00_forms
+parent: form_validation
 ---
 
 Modern browsers understand when a form is incorrectly submitted. The constraint
 validation API allows use to take advantage of this to style our own forms in
-the way we see fit.
+the way we see fit. The contraint validation API is built-in to modern browsers
+and is incredibly comprehensive.
+
+An example of when you might want to do this, a required input can have the required
+attribute added to it:
+
+<code><input type="text" name="needed-input" required></code>
+
+You can also use the HTML5 valid input types and those will be checked as well:
+
+<code><input type="email" name="email"></code>
+
+To see all of the options available (they are ever-growing), visit the
+<a href="https://developer.mozilla.org/en-US/docs/Web/Guide/HTML/HTML5/Constraint_validation">
+MDN site on this topic</a>.
+
+We can also add custom validation to elements! In order to do this, code will need
+to be added to the file <code>src/pivotal-ui/components/forms/validator.js</code>.
+
+Example, our link updater module:
+
+```js_example
+if(isLinkUpdater) {
+  var err = isInvalidName(input.value);
+  if(err) {
+    input.setCustomValidity(err.message);
+  } else {
+    input.setCustomValidity('');
+  }
+}
+```
+
+Setting the custom validity to a value will mean you're informing the browser that
+there is indeed an error here. If you set the value to an empty string, that means
+that there no longer is an error.
+
+To use this validation on your form, add <code>data-validate</code> to the form element.
```

Comment:
So, the custom validation does not force any text to do something. All it does is provide an error if the input does not meet the requirements.